webapp/test_results_handler.go

```diff
@@ -106,7 +106,7 @@ func getTestRunsAndSources(r *http.Request, runSHA string) (testRunSources []str
 			return nil, nil, errors.New("before param provided, but after param missing")
 		}
 
-		const singleRunURL = `/api/run?sha=%s&browser=%s`
+		const singleRunURL = `/api/run?sha=%s&product=%s`
```

Comment:
Not sure if this is the place, but can you add some tests for the combination of `product`, `products`, `browser` and `browsers` params?

Reply:
Done.